addons/mail/models/mail_address_mixin.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError
+
+
+class MailAddressMixin(models.AbstractModel):
+    """ This Mixin can be used to store a normalized version of the primary email address of the parent model.
+        The email_normalized field can than be used on that model to search quickly on emails (by simple comparison
+        and not using time consuming regex anymore).
+        USAGE : the field '_primary_email' must be overridden in the model that inherit the mixin
+        and must contain the email field of the model.
+        """
+    _name = 'mail.address.mixin'
+    _description = 'Email address mixin'
+    _primary_email = ['email']
```

Comment:
Why an array if we force it to have a length of 1 (in _assert_primary_email). Or we should suppose multiple email fields to be normalized on the same model ...

Reply:
done in one of the next commits